### DISCUSSION TITLE
ERROR: No matching distribution found for torch==2.1.0

### DISCUSSION THREAD

**Discussion by sumeet030990** [lllyasviel/Fooocus#3358] [Q&A] [answered]
Trying to install Fooocus on my machine, but getting error
`ERROR: Could not find a version that satisfies the requirement torch==2.1.0 (from versions: 2.2.0, 2.2.1, 2.2.2, 2.3.0, 2.3.1, 2.4.0)
ERROR: No matching distribution found for torch==2.1.0`

I also tried commands such as
`pip3 install torch==2.1.0`

but still getting error

Can someone please help

**Comment by mashb1t** [comment]
Ensure that you've enabled your venv / conda env / pyenv etc. and you're using Python 3.10, then run the command `pip install torch==2.1.0 torchvision==0.16.0 --extra-index-url https://download.pytorch.org/whl/cu121`.

**Reply by sumeet030990** [reply]
ran this command also, but getting same error

**Reply by mashb1t** [reply, marked as answer]
@sumeet030990 you're still using Python 3.12, which, for the third time now, is not, i repeat, IS NOT, compatible with Fooocus without adjustments...

Follow https://github.com/lllyasviel/Fooocus/discussions/3358#discussioncomment-10150565 for further instructions on how to adjust the code to add compatibility with python 3.12.

**Comment by sumeet030990** [comment]
![image](https://github.com/user-attachments/assets/c5c77696-33ac-47de-ba13-5313fc826e45)

Activated the env using
conda activate fooocus
 and then ran 
`python entry_with_update.py` and getting the above error

python version I am using is 3.12.3

**Reply by mashb1t** [reply]
> Ensure that [...] you're using Python 3.10

Fooocus is not compatible with 3.12 (yet), which is why the readme and me both mention to use 3.10.
Feel free to adjust everything according to https://github.com/lllyasviel/Fooocus/pull/2927#issuecomment-2231101766 and the PR content yourself.
